qemu/tests/ctrl_vlan.py

```diff
@@ -0,0 +1,73 @@
+import logging
+
+from virttest import error_context
+from virttest import utils_net
+from virttest import utils_test
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Run ctrl_vlan check test.
+
+    1) Boot vm with ctrl_vlan=on/off
+    2) Verify if netkvm.sys is enabled in guest(only windows)
+    3) Check vlan table in rx-filter information
+    4) If ctrl_vlan=on, do step 5-6
+    5) Set vlan in guest
+    6) Check vlan table in rx-filter information again
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def verify_vlan_table(expect_vlan=None):
+        error_context.context("Check vlan table in rx-filter", logging.info)
+        query_cmd = "query-rx-filter name=%s" % vm.virtnet[0].device_id
+        vlan_table = vm.monitor.send_args_cmd(query_cmd)[0].get("vlan-table")
+        if not expect_vlan:
+            vlan_table.sort()
+            if len(set(vlan_table)) == 4096 and vlan_table[0] == 0 and vlan_table[-1] == 4095:
```

Comment:
Have you tested this patch on RHEL.8.0 guest? The output before vlan set should be vlan-table[], no value anymore. Please check it again. Thanks.

Reply:
Moreover, after add a vlan to guest, vlan-table becomes [1, 0].
After remove the vlan, vlan-table becomes [0].
And if ctrl_vlan = off, vlan-table filled with [4095, 4094, 4093, ..., 0].

Reply:
Moreover, after add a vlan to guest, vlan-table becomes [1, 0].
After remove the vlan, vlan-table becomes [0].
And if ctrl_vlan = off, vlan-table filled with [4095, 4094, 4093, ..., 0].

As for rhel7.6, with ctrl_vlan=on
1). before set vlan in guest, "vlan-table": [0] (different from rhel8.0 output)
2). after set vlan in guest, "vlan-table": [1, 0] (the same as rhel8.0)
3). after remove vlan in guest, "vlan-table": [0] (the same as rhel8.0)
and with ctrl_vlan=off, the result is as the same of rhel8.0

Reply:
@xiywang I writed the case according to the polarion case, if the behaviour is different on rhel8, I suggest to update the polarion case first.

Reply:
Updated the expected results in polarion.

Reply:
@xiywang The result is weird for rhel8 guest, the vlan-table should not be `null list` when we already add a vlan for it, could you confirm if it is not a product bug?

Reply:
It isn't null. It's [1, 0] after add a vlan. Please check the expected result after step 5 in polarion.

Reply:
@xiywang sorry, I mean before we add a vlan it should not be `non list`